### PR TITLE
Rename pathlib hook parameters 

### LINKED ISSUE #9283: pytest 7: path vs. fspath, pathlib vs. py.path
I've just been looking through the changelog for pytest 7 and noticed something I'd like to discuss before the release:

In #8144, @bluetech added `pathlib` alternatives to various hooks, and called them `fspath`, with the old `py.path.local` alternatives keeping their `path` name.

However, in #8251, @RonnyPfannschmidt added a `pathlib` attribute to `Node`, and called it `path`, with the old `py.path` one being called `fspath`...

We then reverted the `Node.fspath` deprecation in #8903 after some discussion in #8821, but kept the new `path` attribute... 

However, the naming inconsistency seems like something that could confuse people. What to do? Keep it as is, or revert the new attribute entirely until we figured out the rest of the deprecation? Make it private perhaps?

### PR DESCRIPTION
**Pull request by nicoddemus**
Fixes #9283

### CODE REVIEW

File: testing/plugins_integration/requirements.txt [hunk 1]
```diff
@@ -4,7 +4,9 @@ pytest-asyncio==0.16.0
 pytest-bdd==5.0.0
 pytest-cov==3.0.0
 pytest-django==4.4.0
-pytest-flakes==4.0.4
+# See https://github.com/asmeurer/pytest-flakes/pull/45
```
Comment (nicoddemus) [inline testing/plugins_integration/requirements.txt:7]: Needed a new PR on `pytest-flakes`. 👍

**Review by bluetech — APPROVED** [review]

**Review by RonnyPfannschmidt — APPROVED** [review]
its a bit unfortunate the names got longer but it may serve users better after all :+1:

File: testing/plugins_integration/requirements.txt (hunk 1 above)
Comment (The-Compiler) [reply testing/plugins_integration/requirements.txt:7]: Given that @asmeurer plans a release according to https://github.com/asmeurer/pytest-flakes/pull/45#issuecomment-984171700, should we wait for that, so we can use `==4.0.5` or whatever here instead?

Comment (nicoddemus) [reply testing/plugins_integration/requirements.txt:7]: Yeah that's my plan. 👍

Comment (nicoddemus) [reply testing/plugins_integration/requirements.txt]: Done!

**Review by The-Compiler — APPROVED** [review]
Looking great, thanks!

**Pull request merged**